Merchant/Core/Services/SocketService.swift

```diff
@@ -53,15 +54,11 @@ class SocketService {
             let message = message as? String
             let data = message?.data(using: .utf8)
             
-            guard let transaction = try? JSONDecoder().decode(WebSocketTransactionResponse.self, from: data!) else {
-                return
+            if let transaction = try? JSONDecoder().decode(WebSocketTransactionResponse.self, from: data!) {
```

Comment:
@Mattijah **[IMO]**
I know it was in the previous code too, but here data should not be forced unwrap. Please use guard instead.

Reply:
Yes, well found.

Reply:
I agree with you guys, but again this commit was about something else. I didn't really wanna fix forced unwrapping as a part of commit that deals with something else. If I was then ask to revert this commit (which changes guards into ifs) I would then also undo the fix **or** I would need to go and revert all changes manually. It doesn't make any sense. I hope you understand me.

Reply:
It’s a PR. We are not here to point fingers on who did what but to improve the project. He pointed out you an issue. I think this problem as well as fixing the leak that I wrote with the web socket are more relevant than changing every single var to let or removing comment.

Reply:
Sorry, I thought "_Please use guard instead_" was directed to me.

> I think removing a force wrap as well as fixing the leak that I wrote with the web socket are more relevant than changing every single var to let or removing comment.

- The leak is fixed
- There were probably 20x more properties with incorrect type than force unwrap. I simply noticed it first and correct it. Apart from that I did another 12 changes I can't fix it all. There is still plenty of stuff left.